compiler/resolution/functionResolution.cpp

```diff
@@ -14679,7 +14679,8 @@ void handleDefaultAssociativeWarnings(Symbol* sym,
 
             USR_WARN(sym, "The default parSafe mode for associative domains "
                      "and arrays (like '%s') is changing from 'true' to "
-                     "'false'. To suppress this warning, use an explicit "
+                     "'false'. To suppress this warning you can make your "
+                     "domain const, use an explicit "
```

Comment:
Given that is warning is so long, you could consider splitting it up and using `USR_WARN` followed by `USR_PRINT`

Reply:
That's a great idea! I hadn't considered that. What if the warning looked like this:
```
test/unstable/assoc-domain-parsafe-changed.chpl:41: warning: The default parSafe mode for associative domains and arrays (like 'aa') is changing from 'true' to 'false'.
test/unstable/assoc-domain-parsafe-changed.chpl:41: note: To suppress this warning you can make your domain const, or
test/unstable/assoc-domain-parsafe-changed.chpl:41: note: use an explicit parSafe argument (ex: domain(int, parSafe=false)), or
test/unstable/assoc-domain-parsafe-changed.chpl:41: note: compile with '-snoParSafeWarning'. 
test/unstable/assoc-domain-parsafe-changed.chpl:41: note: To use the old default of parSafe=true, compile with '-sassocParSafeDefault=true'.

```
This might be too many lines. Maybe I can combine lines 2,3, and 4? Or maybe it looks better this way?

Reply:
I would probably merge lines 2, 3, and 4, although I don't have a problem with it as-is

```
To suppress this warning mark the domain as const, use an explicit parSafe argument (ex: domain(int, parSafe=false)), or compile with '-snoParSafeWarning'
```